libraries/tools/kotlin-gradle-plugin/src/main/kotlin/org/jetbrains/kotlin/gradle/kpm/idea/IdeaKotlinProjectModelBuilder.kt

```diff
@@ -0,0 +1,24 @@
+package org.jetbrains.kotlin.gradle.kpm.idea
+
+import org.gradle.api.Project
+import org.gradle.tooling.provider.model.ToolingModelBuilder
+import org.jetbrains.kotlin.compilerRunner.konanHome
+import org.jetbrains.kotlin.gradle.dsl.pm20Extension
+import org.jetbrains.kotlin.gradle.plugin.mpp.pm20.KotlinPm20ProjectExtension
+import java.io.File
+
+class IdeaKotlinProjectModelBuilder : ToolingModelBuilder {
+    override fun canBuild(modelName: String): Boolean = modelName == IdeaKotlinProjectModel::class.java.name
+    override fun buildAll(modelName: String, project: Project): IdeaKotlinProjectModel {
+        return project.pm20Extension.toIdeaKotlinProjectModel()
+    }
+}
+
+internal fun KotlinPm20ProjectExtension.toIdeaKotlinProjectModel(): IdeaKotlinProjectModel {
+    return IdeaKotlinProjectModelImpl(
+        coreLibrariesVersion = coreLibrariesVersion,
+        explicitApiModeCliOption = explicitApi?.cliOption,
+        kotlinNativeHome = File(project.konanHome).absoluteFile,
+        modules = modules.map { module -> module.toIdeaKotlinModule() }
```

Comment:
Unrelated directly to this PR, but there could be multiple modules in a single KMP project? My understanding has always been there is only 1 module, and many fragments, some of which are variant fragments.

Reply:
There can be arbitrary modules in a project!
The default is to have at least two modules
`main` -> fragments + variants
`test` -> fragments + variants

test will be declaring a module dependency on main
test will be marked a friend module of main

Reply:
Android therefore could create a new module called `instrumentedTest` where e.g. only android related variants and fragments will be registered by default!